doc/reference/trd-appid.md

```diff
@@ -0,0 +1,396 @@
+application and know whether a particular userspace binary belongs to
+an application. 
+
+The mapping between binaries and applications can be 
+many-to-many. Multiple binaries can be associated with a single application
+when there are software updates/versions or when an application needs
+to run in multiple processes. A program that migrates data from one
+application to another (e.g., transitions keys from an old U2F
+application to a new one) needs to be associated with both the source
+and destination applications.
+
+To remain flexible and support many use cases, the Tock kernel makes 
+minimal assumptions on the structure and form of
+application credentials that bind an application identifier to a
+binary. Application credentials are arbitrary k-byte sequences that
+are stored in an userspace binary's Tock binary format (TBF)
+headers. When a Tock board instantiates the kernel, it passes a
+reference to an AppID (application identifier) verifier, which the
+kernel uses to determine the AppIDs of each userspace binary it reads
+and decide whether to load the binary into a process.
+
+Most of the complications in AppIDs stem from the fact that they are a
+general mechanism used for many different use cases. Therefore, the
+exact structure and semantics of application credentials can vary
+widely. Tock's TBF header formats, kernel interfaces and mechanisms
+must accomodate this wide range.
+
+The interfaces and standard implemenentations for AppIDs and AppID
+verifiers are in the kernel crate, in module `appid`. There are two
+main traits:
+
+  * `kernel::appid::Verifier` is responsible for defining which types
+  of application identifier it can accepts and whether it accepts a
+  particular application identifier for a specific application
+  binary. The kernel only loads application binaries the `Verifier`
+  accepts.
+  
+  * `kernel::appid::Compress` is repsonsible for compressing
+  application identifiers into short, 32-bit identifiers called
+  `ShortID`s. `ShortID`s provide a mechanism for fast comparison,
+  e.g., for an application identifier against an access control list.
+  
+
+2 Terminology
+===============================
+
+This document uses several terms in precise ways. Because these terms
+overlap somewhat with general terminology in the Tock kernel, this
+section defines them for clarity. The Tock kernel often uses the term
+"application" to refer to what this document calls a "Tock binary."
+
+**Application**: userspace software developed and maintained by an
+individual, group, corporation, or organization that meets the
+requirements of a particular Tock device use case. An application can
+consist of a single userspace binary, multiple userspace binaries that
+run concurrently (an application split into several processes), or
+multiple userspace binaries only one of which runs at a time (an
+application with multiple versions).
+
+**Tock binary**: a Tock binary format (TBF)[TBF] object stored on a
```

Comment:
Wouldn't this be "Tock userspace binary"?

Reply:
I thought we had agreed on "process binary" in a core WG call.